### DISCUSSION TITLE
table和treeTable单元格编辑功能疑问

### DISCUSSION THREAD

**Discussion by hjingsuper** [layui/layui#1811] [讨论]
示例：[https://codepen.io/hjingsuper/pen/qBwQVxz](https://codepen.io/hjingsuper/pen/qBwQVxz)

> 1.单元格采用函数写法开启编辑时，每次点击编辑都会触发函数。单元格能否编辑，不是在初次渲染就已经确定了吗？就算修改其他字段时更新了条件，也只是重新渲染1次吧。不太理解为什么每次点击编辑都会去触发函数

> 2.treeTable删除后，出现字段丢失bug（不知道怎么描述）。内部缓存未同步bug的issues已关闭代码已经合并，下载github最新代码也会出现

操作步骤：
1. 初始化数据
2. 新增主键 -> 编辑【用户名 321】
3. 删除【用户名 1】
4. 此时再次编辑【用户名 321】出现错误

![2024-04-20-16-40-39](https://github.com/layui/layui/assets/36070786/11ebad1b-3707-491d-a677-712ee52ce5b7)

**Comment by Sight-wcg** [comment]
>1 不太理解为什么每次点击编辑都会去触发函数

可能是因为兼容性，edit 函数类型是 2.7 增加的

> 2.treeTable删除后，出现字段丢失bug（不知道怎么描述）。内部缓存未同步bug的issues已关闭代码已经合并，下载github最新代码也会出现

确认是 BUG, 已定位，具体原因见注释 https://github.com/layui/layui/pull/1760/files#diff-13952582bff0a1aa2ddde6f94079a336fc5ff89fbb220988854ebf97ca7f6b20L2288 

你在 issues 基于这个讨论重新创建一个问题，我会尽快修复。

![image](https://github.com/layui/layui/assets/26325820/42b16993-1453-4e3b-beba-b59deee15cd2)
